### PR TITLE
spanner: use joda LocalDate

### LINKED ISSUE #3157: com.google.cloud.Date has no interop w/ java.util.Date
It's very hard to convert c.g.c.Date to j.u.Date. Can c.g.c.Date add something similar to `toDate`, `fromDate` method?

### PR DESCRIPTION
**Pull request by pongad**
This also lets us nuke com.google.cloud.Date and fixes #3157.

### CODE REVIEW

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java [hunk 1]
```diff
@@ -19,9 +19,9 @@
 import static com.google.common.base.Preconditions.checkState;
 
 import com.google.cloud.ByteArray;
-import com.google.cloud.Date;
 import com.google.cloud.Timestamp;
 import java.util.List;
+import org.joda.time.LocalDate;
```
Comment (garrettjonesgoogle) [inline google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java]: We should be using date types from threetenbp, not joda.

Comment (pongad) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java]: Oh! I saw we're using Joda in a few places, but those are internal. Will fix this.

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by vkedia** [comment]
This is a breaking change and will likely break many clients. Why do we need to do this especially since we will eventually want to use java 8 LocalDate here?

**Comment by pongad** [comment]
@vkedia Hm, I thought that this was something you wanted to do but didn't have time to do it. We can close this if you'd like.

**Comment by vkedia** [comment]
No we want to move to java 8 LocalDate at some point but not to this.

**Comment by vkedia** [comment]
cc @snehashah16

**Pull request closed without merging**